cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp

```diff
@@ -38,8 +43,39 @@ template <class Fr, size_t _length> class Univariate {
         }
     }
 
+    // TODO(luke): I'd like to be able to construct a Univariate directly from a buffer this leads to ambiguity with the
+    // 'evaluations' based constructor above. Could find a way to resolve if desired.
+
+    // explicit Univariate(std::vector<uint8_t> buffer)
+    //     : evaluations({ { 0 } })
+    // {
+    //     const size_t num_elements = buffer.size() / sizeof(Fr);
+    //     ASSERT(num_elements == _length);
+    //     for (size_t i = 0; i < num_elements; ++i) {
+    //         evaluations[i] = from_buffer<Fr>(buffer, i * sizeof(Fr));
+    //     }
+    // }
+
     Fr& value_at(size_t i) { return evaluations[i]; };
 
+    // Write the Univariate evaluations to a buffer
+    std::vector<uint8_t> to_buffer() const
```

Comment:
Did you try to implement in terms of the prescription [here](https://github.com/AztecProtocol/barretenberg/blob/ab331f74c3a00ae0babd531ecb3d300365aae483/cpp/src/aztec/common/serialize.hpp#L9) and run into trouble?

Reply:
...oh, we put stuff in the `std` namespace?? TODO: are we violating https://en.cppreference.com/w/cpp/language/extending_std?